src/libexpr/primops/meson.build

```diff
@@ -0,0 +1,27 @@
+# Nix lib expr build file
+#============================================================================
+
+
+libexpr_priomops_dir = meson.current_source_dir()
+
+# src files
+#============================================================================
+
+libexpr_inc += include_directories('.')
+
+libexpr_src += files(
+    join_paths(libexpr_priomops_dir, 'context.cc'),
```

Comment:
I think you are going to need a new enough version of meson that strings have the `/` operator, so you cand do:
```meson
libexpr_priomps_dir / 'context.cc',
```
if you prefer.

Reply:
whatever is more efficient / you think is best :)

Reply:
It's really a style choice, under the hood Meson does exactly the same thing.

Reply:
i think join_paths is more verbose and easier to understand, but / isnt that complicated and theres enough context to where it doesnt really matter. personally i don't mind, but it would save a ton of typing

Reply:
@dcbaker the latest stable version of Meson does support all the appointments you have made in this review?

Reply:
since this `meson.build` is in the `libexpr_priomops_dir`, i thought none of that was needed?

Reply:
Oh, you're right, just `files('context.cc')` should be sufficient, Meson assumes a bare files() argument is in the current source dir.

Reply:
i dont remember what the added path root was for, but again it was to do with problems on *BSD :)

Reply:
My opinion: ` join_paths()` looks like a "my language has no support for concatenate paths, then I created this function that behind the scenes adjust the paths, oh those Windows quirks..." - indeed it remembers me cmake.

Using `/` looks more natural.

And it reduces 10 characters!

Reply:
I think for cases like this we can always come back for the BSDs later.  (Though I also suspect something else was afoot and it's not really a problem there.) Bare path is best.